Src/FluentAssertions/Primitives/DateTimeOffsetAssertions.cs

```diff
@@ -78,11 +188,11 @@ public DateTimeOffsetAssertions(DateTimeOffset? value)
         /// <param name="becauseArgs">
         /// Zero or more objects to format using the placeholders in <paramref name="because" />.
         /// </param>
-        public AndConstraint<TAssertions> Be(DateTimeOffset? expected, string because = "",
+        public AndConstraint<TAssertions> BeExactly(DateTimeOffset? expected, string because = "",
             params object[] becauseArgs)
         {
             Execute.Assertion
-                .ForCondition(Subject == expected)
+                .ForCondition((Subject == null && expected == null) || (Subject != null && expected != null && Subject.Value.EqualsExact(expected.Value)))
```

Comment:
We strive to keep lines at atmost 130 characters wide.
```suggestion
                .ForCondition((Subject == null && expected == null) 
                    || (Subject != null && expected != null && Subject.Value.EqualsExact(expected.Value)))
```